### ISSUE TITLE
lookup plugin (bigiq_license) not found

### ISSUE THREAD

**Issue by onurbrc** [F5Networks/f5-ansible#1952]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
```
bigiq_regkey_license
```

##### ANSIBLE VERSION
```
Ansible 2.10.6
Collections 1.7.1
```

##### PYTHON VERSION
```
python3.8
```

##### BIGIQ VERSION

```
BIG-IQ Virtual Edition  7.1.0.2-0.0.21
```

##### CONFIGURATION
```
[defaults]
roles_path           = ./roles       # ANSIBLE_ROLES_PATH 
inventory            = ./inventory   # ANSIBLE_INVENTORY
collections_path     = ~/.ansible/collections     # ANSIBLE_COLLECTIONS_PATH
force_color          = 1
gathering            = smart
callback_whitelist   = profile_tasks
deprecation_warnings = False
command_warnings     = False
host_key_checking    = False

[ssh_connection]
ssh_args  = -o UserKnownHostsFile=/dev/null -o ForwardAgent=yes -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no

[inventory]
enable_plugins=generator,host_list,script,yaml,ini

```

##### SUMMARY
Trying to select a random license key from a pool of biqiq available licenses.

##### STEPS TO REPRODUCE

Using examples: https://docs.ansible.com/ansible/latest/collections/f5networks/f5_modules/bigiq_license_lookup.html#examples

```
  tasks:
    - name: Get a regkey license from a license pool
      bigiq_regkey_license:
        key: "{{ lookup('bigiq_license', pool_name='foo_pool', username=baz, password=bar, host=192.168.1.1, port=443}}"
        state: present
        pool: foo_pool
       register: __output

    - name: Show result
      debug:
        msg: "{{ __output }}"
```

##### EXPECTED RESULTS
Success getting information 

##### ACTUAL RESULTS
```
fatal: [192.168.1.1]: FAILED! => {
    "msg": "lookup plugin (bigiq_license) not found"
}
```

**Comment by trinaths** [comment]
@onurbrc - Please share the playbook where your are getting this error.

**Comment by onurbrc** [comment]
@trinaths 

~~~yaml
- name: Check and select next free license
  hosts: BIGIQ
  connection: local

  tasks:
    - name: Get a regkey license from a license pool
      f5networks.f5_modules.bigiq_regkey_license:
        key: "{{ lookup('bigiq_license', pool_name='myLicencePool', username=admin, password=admin, host=10.28.01.21, port=443) }}"
        state: present
        pool: myLicencePool
      register: __output

    - name: Show result
      debug:
        msg: "{{ __output }}"
~~~

**Comment by focrensh** [comment]
Are you able to try the examples from the latest docs in:

https://clouddocs.f5.com/products/orchestration/ansible/devel/modules/bigiq_regkey_license_module.html#bigiq-regkey-license-module


If you are trying to assign the license to a BIG-IP, this is the module for that: https://clouddocs.f5.com/products/orchestration/ansible/devel/modules/bigiq_regkey_license_assignment_module.html#bigiq-regkey-license-assignment-module

**Comment by onurbrc** [comment]
@focrensh I'm trying to use the recent plugin which tells me that it can select a random license key from a pool of biqiq available licenses

source: https://docs.ansible.com/ansible/latest/collections/f5networks/f5_modules/bigiq_license_lookup.html#ansible-collections-f5networks-f5-modules-bigiq-license-lookup

In my case, if I have many  licences in my pool I just want to find the reference of the next one that is free.

**Comment by wojtek0806** [comment]
@onurbrc specify `collection: f5networks.f5_modules` in the playbook level not task level and see if this works

**Comment by onurbrc** [comment]
@wojtek0806 Actually in the `STEPS TO REPRODUCE` section I'm precising the tasks without de collection_path concatenated. So, in both ways it is not working for me. 

Do you need any additional information about my environment, connection plugin, delegate_to usage, etc? 

This lookup would be very useful for me.

**Comment by trinaths** [comment]
Created INFRAANO-202 for PM tracking

**Comment by wojtek0806** [comment]
this will be fixed in 1.9.0 work merged in devel
